### ISSUE TITLE
Possible bad parsing of encoded angle brackets

### ISSUE THREAD

**Issue by thisgeek** [Thibaut/devdocs#325]
This text in [Angular's $sce documentation][angular-sce] (version 1.4.7):

> Note: When enabled (the default), IE<11 in quirks mode is not supported. In this mode, IE<11 allow one to execute arbitrary javascript by the use of the expression() syntax. Refer to learn more about them. 

[angular-sce]: https://code.angularjs.org/1.4.7/docs/api/ng/service/$sce

Does not match this text in the [devdocs $sce documentation][devdocs-sce]:

> Note: When enabled (the default), IE to learn more about them.

[devdocs-sce]: http://devdocs.io/angular/ng/service/%24sce#show-me-an-example-using-sce-.html

My guess is that the web scraper misinterpreted the meaning of the encoded open angle bracket in "IE<11" and what looks like a bogus tag in the source HTML. Here's what angular's HTML looks like:

```
<p>Note:  When enabled (the default), IE&lt;11 in quirks mode is not supported.  In this mode, IE&lt;11 allow
one to execute arbitrary javascript by the use of the expression() syntax.  Refer
<http: blogs.msdn.com="" b="" ie="" archive="" 2008="" 10="" 16="" ending-expressions.aspx=""> to learn more about them.
You can ensure your document is in standards mode and not quirks mode by adding <code><span class="dec">&lt;!doctype html&gt;</span></code>
to the top of your HTML document.</http:></p>
```

The "bogus" tag I mentioned is `<http: blogs.msdn.com="" b="" ie="" archive="" 2008="" 10="" 16="" ending-expressions.aspx="">`. It looks like a mangled link. (See the [source](https://github.com/angular/angular.js/blob/v1.4.7/src/ng/sce.js#L417).)

Here is what the HTML from the currently deployed dev docs looks like:

```
<p>Note: When enabled (the default), IE to learn more about them. You can ensure your document is in standards mode and not quirks mode by adding <code>&lt;!doctype html&gt;</code> to the top of your HTML document.</p>
```

I bet the scraper probably should not interpret `&lt;` as the start of an HTML tag. Maybe it's an upstream bug.

Edit: fixed incorrect documention links.

**Comment by thisgeek** [comment]
I looked over some of the open Nokogiri issues. Nothing jumped out at me, but I am new to the library. sparklemotion/nokogiri/issues/1406 and sparklemotion/nokogiri/issues/1294 _might_ be related.

You, dear reader, would find a place in my heart for helping make a test case that shows whether Nokogiri has a bug that would explain the behavior reported above. I will hazard a guess that the version used to produce the currently deployed devdocs.io is 1.6.7.rc3 according to the [latest Gemfile](https://github.com/Thibaut/devdocs/blob/87080c24ea13935dda640e5161b10b090268e55d/Gemfile#L33).

**Comment by Thibaut** [comment]
> Here's what angular's HTML looks like:

Actually Angular's HTML is at fault, the `<` are not escaped:
https://docs.angularjs.org/partials/api/ng/service/$sce.html

![49](https://cloud.githubusercontent.com/assets/17579/12216958/7215dd14-b6c0-11e5-8eb9-d11602f4b7b7.png)

Browsers handle/repair this correctly but Nokogiri/libxml2 doesn't.

Would you mind sending a PR to Angular to escape the two `IE<11` please?

**Comment by thisgeek** [comment]
@Thibaut You're correct. I must've gotten my samples mixed up.

I'll open a fix in angular. Thank you.

**Comment by Thibaut** [comment]
Closing as this isn't an issue with DevDocs.

**Comment by thisgeek** [comment]
For posterity:

I traced the source of the issue to a probable bug in the [marked](https://github.com/chjj/marked) library, on which angular's documentation build task depends to parse markdown. In my view, the fault is not due to a failure to escape the angle brackets. The [markdown dingus](http://daringfireball.net/projects/markdown/dingus), for example, when given the same input escapes the angle brackets as I expect. I'll look into writing a fix to marked.

**Comment by thisgeek** [comment]
Patched chjj/marked#814
